### PR TITLE
Dev/1.0.1

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

File: src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java [hunk 1]
```diff
@@ -716,7 +716,15 @@ public static void installFirmware_515(
               + ") from "
               + firmwareUrl);
       try {
-        downloadFirmwareFile(firmwareUrl);
+        // Perform firmware download and store result if the download/update was unsuccessful
```
Comment (it-hms) [inline src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java:719]: This comment makes it sound like a success should return.

File: src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java [hunk 2]
```diff
@@ -716,7 +716,15 @@ public static void installFirmware_515(
...
+        String result = downloadFirmwareFile(firmwareUrl);
+
+        // Update state to failed with result (device reboots instead of returning when successful)
```
Comment (it-hms) [inline src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java:722]: This comment makes it sound like the previous method only returns for failure.

**Review by it-hms — DISMISSED** [review]
I approve, but just change one comment. They appear to contradict eachother.

File: src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java (hunk 2 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java:722]: It does only return on failure. If the firmware update was successful, the device halts the JVM and it never returns.

**Review by oliver-walker-hms — APPROVED** [review]

File: src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java]: Updated comment. I agree that it was not described well enough what happens with successful firmware downloads.

**Comment by alexjhawk** [comment]
@it-hms @oliver-walker-hms Made a small patch based on Ian's comment above. For your convenience, here is an exact diff of the changes I've between your previous approval and now: 
image

File: src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java (hunk 1 above)
Comment (it-hms) [reply src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java]: ok.. thanks. Approved!

**Review by it-hms — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**